### ISSUE TITLE
Unexpected visualization results when running test_example.py without sensored_depth_maps

### ISSUE THREAD

**Issue by ChenyiXu04** [jayLEE0301/TraceGen#11]
Hello, thank you for your great work!

I am trying to verify the output of ` test_example.py `, but the visualization results look quite different from what I expected. I would like to confirm whether my setup is correct.

Command

I run the script using the following command:

```
CUDA_VISIBLE_DEVICES=1 python test_example.py \
  --config cfg/eval.yaml \
  --resume checkpoints/tracegen_model.pth \
  --test /data4/xcy_data/TraceForge/output/run_20251102_151010 \
  --output results/test_run \
  --guidance_scale 2.0
```
The script outputs 10 visualization images.

However, I found that the generated results vary significantly and the visual quality is not as good as expected.

Model Weights

The model weights I used were downloaded from the official HuggingFace repository:

` JayLee131 / TraceGen_Libero `

Dataset

The test data comes from the Quick test dataset mentioned in the TraceForge repository, and it was further processed using the TraceForge pipeline.

The resulting dataset structure contains:
```
images

depth_maps

label.json
```

However, this dataset does not include `sensored_depth_maps `.

For ` label.json `, I copied the first instruction from ` three_instructions.json `.
```
images

depth_maps

label.json
```
However, in this dataset there are no ` sensored_depth_maps ` provided.

For ` label.json `, I simply copied the first instruction from ` three_instructions.json `.

Handling Missing ` sensored_depth_maps `

In ` test_example.py `, the code expects ` sensored_depth_maps ` as input to compute the depth ratio.

Since my dataset only contains depth_maps, I modified the code to use a fallback value:
```
depth_ratio_filled = np.ones((images.shape[2], images.shape[3]), dtype=np.float64)
```
This means the model effectively uses the original depth_maps without additional sensor correction.

Questions

Is it expected that the visualization results differ significantly when ` sensored_depth_maps ` are not provided?

Is setting depth_ratio_filled to all ones a reasonable fallback strategy, or could this significantly affect the final rendering results?

When using TraceForge-generated data, what is the recommended way to run ` test_example.py ` if ` sensored_depth_maps ` are unavailable?

Any clarification would be greatly appreciated. Thank you!

### Example Outputs

Below are some of the generated visualization results.

<img width="970" height="1029" alt="Image" src="https://github.com/user-attachments/assets/6f26a80f-ff34-4b5a-9e0f-ac11eca460d0" />

<img width="970" height="1029" alt="Image" src="https://github.com/user-attachments/assets/0ea2d03b-cab2-448b-937f-db0fb7619d28" />

<img width="970" height="1029" alt="Image" src="https://github.com/user-attachments/assets/15b358d5-27b0-49d5-914b-d32dae0d714b" />

**Comment by jayLEE0301** [comment]
Hello @ChenyiXu04 

Thank you for your interest in TraceGen and for providing such a detailed report on your setup! Regarding your questions about the depth ratio and visualization quality, here are some clarifications:

1. Handling Missing `sensored_depth_maps`

Your approach of setting `depth_ratio_filled` to all ones is reasonable for visualization and simulation-based testing.

In our pipeline, the depth ratio is primarily used as a post-processing scaling factor to align the predicted $z$ values with the physical scale of a specific real-world depth sensor. Thus, it does not change the core structure of the generated trajectory or the visualization results. You can safely proceed with your current fallback strategy if you are not running real-world experiments.

2. Visualization Quality and Domain Shift

The discrepancy you are seeing in the visualization is likely due to a domain shift between the training and testing environments, rather than the depth ratio implementation.

* Model Design TraceGen is designed as a lightweight and efficient model. While this makes it highly adaptable, it also means you need few (5~15) task relevant demo (could be different embodiment or background, but similar trajectory or behavior) to *warm-up* the model.
* The weights you downloaded (`TraceGen_Libero`) were fine-tuned specifically on the LIBERO dataset. The visual features, camera angles, and object distributions in LIBERO are quite distinct from the test image you are using.

3. Recommended Workflow for New Datasets

To get the best performance when applying TraceGen to a new data distribution like your TraceForge setup, we recommend the following:

* Few-shot Warm-up: As mentioned in our paper, we highly recommend performing a **5–15 shot "warm-up" (fine-tuning)** using demonstrations that are task-relevant to your specific setup.

We hope this clarifies the behavior you're seeing! Please let us know if you have any further questions as you experiment with the warm-up process.

**Comment by ChenyiXu04** [comment]
> Hello [@ChenyiXu04](https://github.com/ChenyiXu04)
> 
> Thank you for your interest in TraceGen and for providing such a detailed report on your setup! Regarding your questions about the depth ratio and visualization quality, here are some clarifications:
> 
> 1. Handling Missing `sensored_depth_maps`
> 
> Your approach of setting `depth_ratio_filled` to all ones is reasonable for visualization and simulation-based testing.
> 
> In our pipeline, the depth ratio is primarily used as a post-processing scaling factor to align the predicted z values with the physical scale of a specific real-world depth sensor. Thus, it does not change the core structure of the generated trajectory or the visualization results. You can safely proceed with your current fallback strategy if you are not running real-world experiments.
> 
> 2. Visualization Quality and Domain Shift
> 
> The discrepancy you are seeing in the visualization is likely due to a domain shift between the training and testing environments, rather than the depth ratio implementation.
> 
> * Model Design TraceGen is designed as a lightweight and efficient model. While this makes it highly adaptable, it also means you need few (5~15) task relevant demo (could be different embodiment or background, but similar trajectory or behavior) to _warm-up_ the model.
> * The weights you downloaded (`TraceGen_Libero`) were fine-tuned specifically on the LIBERO dataset. The visual features, camera angles, and object distributions in LIBERO are quite distinct from the test image you are using.
> 
> 3. Recommended Workflow for New Datasets
> 
> To get the best performance when applying TraceGen to a new data distribution like your TraceForge setup, we recommend the following:
> 
> * Few-shot Warm-up: As mentioned in our paper, we highly recommend performing a **5–15 shot "warm-up" (fine-tuning)** using demonstrations that are task-relevant to your specific setup.
> 
> We hope this clarifies the behavior you're seeing! Please let us know if you have any further questions as you experiment with the warm-up process.

Thank you for your detailed follow-up and for sharing the updated experimental setup.

I re-ran the evaluation using the **TraceGen_EpicKitchen** weights (downloaded from the Hugging Face repository _JayLee131/TraceGen_EpicKitchen_) directly on the corresponding **TraceGenEpickitchen** dataset, in a zero-shot setting (without additional fine-tuning).

The command I used is:
```
python test_example.py \
  --config /data4/xcy_data/mycode/TraceGen/cfg/eval.yaml \
  --resume /data4/xcy_data/Checkpoints/TraceGen_EpicKitchen/tracegen_epickitchen.pth \
  --test /data4/xcy_data/TraceGenEpickitchen/P09_104_32 \
  --output ./test_result \
  --guidance_scale 2.0 \
  --debug
```
Some example visualization results from this zero-shot evaluation are shown below:

<img width="970" height="1029" alt="Image" src="https://github.com/user-attachments/assets/9f97afa7-3938-4157-8503-d7284f7e14a1" /><img width="970" height="1029" alt="Image" src="https://github.com/user-attachments/assets/f4e79fae-0184-4a05-ad6b-de54bfdb146f" />

Additionally, I performed a small few-shot warm-up (fine-tuning) using approximately 8–12 task-relevant demonstration samples from the same TraceGenEpickitchen dataset, starting from the tracegen_epickitchen.pth checkpoint. The visualization results after this warm-up stage are shown below:

<img width="970" height="1029" alt="Image" src="https://github.com/user-attachments/assets/e7d484f4-dcde-450a-83d3-c0824fc6e186" /><img width="970" height="1029" alt="Image" src="https://github.com/user-attachments/assets/e8ae3bfe-eb08-41c0-8c7f-1a104bb3fb75" />

I would like to confirm whether the following observations are consistent with the expected behavior of TraceGen under these settings.
Thank you again for your time and for your helpful guidance.

Best regards,
Chenyi

**Comment by jayLEE0301** [comment]
Hello @ChenyiXu04 and thank you for sharing the updates!

It is difficult to determine whether these output results align with our expectations for this specific image set. Although the training set includes 'in-the-wild' videos like Epic-Kitchens, our testing was primarily focused on more static scenes within a robot setup.

As a sanity check, it might be helpful to evaluate the prediction results on the training set to ensure your pipeline is functioning correctly.

Regarding the performance on your target task - if the current predictions are not yet sufficient, you may need more in-distribution data or to implement sampling from the generated traces (though this wasn't necessary for our specific setup).

Please let us know if you need to further improve the accuracy for your task, or if you could share the specific level of precision you're aiming for. I look forward to continuing our discussion.

Best,
Seungjae.
